lib/libc/picolibc/CMakeLists.txt

```diff
@@ -1,10 +1,15 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
-zephyr_library_sources(libc-hooks.c)
-
-# Do not allow LTO when compiling libc-hooks.c file
-set_source_files_properties(libc-hooks.c PROPERTIES COMPILE_OPTIONS $<TARGET_PROPERTY:compiler,prohibit_lto>)
```

Comment:
This is not needed anymore?

Reply:
Fair enough. From the commit message it kind of seemed like it may have been an oversight to remove this :)

Reply:
yeah, I hadn't remembered that part of the change as I did this a long time ago. Good to review and make sure nothing got missed.